src/ReaderThemeSupportFeatures.php

```diff
@@ -327,10 +422,20 @@ private function print_editor_font_sizes_styles( array $font_sizes ) {
 			if ( ! $this->has_required_feature_props( self::FEATURE_EDITOR_FONT_SIZES, $font_size ) ) {
 				continue;
 			}
+
+			// Just in case the font size is not in the expected format.
+			$font_size[ self::KEY_SIZE ] = $this->get_typography_value_and_unit( $font_size[ self::KEY_SIZE ] );
+
+			if ( empty( $font_size[ self::KEY_SIZE ] ) ) {
+				continue;
+			}
+
 			printf(
-				':root .is-%1$s-text, :root .has-%1$s-font-size { font-size: %2$spx; }',
+				':root .is-%1$s-text, :root .has-%1$s-font-size { font-size: %2$s; }',
 				sanitize_key( $font_size[ self::KEY_SLUG ] ),
-				(float) $font_size[ self::KEY_SIZE ]
+				function_exists( 'wp_get_typography_font_size_value' )
+					? esc_attr( wp_get_typography_font_size_value( $font_size ) )
+					: esc_attr( $font_size[ self::KEY_SIZE ] )
```

Comment:
Minor point, but I don't think `esc_attr()` is right here. I think `strip_tags()` is the right choice. See https://github.com/WordPress/wordpress-develop/blob/200868214a1ae0a108dac491677ba82e7541fc8d/src/wp-includes/theme.php#L1891-L1896

Reply:
It does make sense, but I am not sure if we will ever get tags here. Or better to never assume and update it to use `strip_tags()` 🤔

Reply:
Better to never assume, yeah

Reply:
Just realized, we will still need the escaping, as `strip_tags` is not escaping a utility.

Reply:
Well, it is somewhat. It's what core uses in `wp_custom_css_cb()`.